enterprise/server/remote_execution/action_merger/action_merger.go

```diff
@@ -117,14 +130,69 @@ func RecordClaimedExecution(ctx context.Context, rdb redis.UniversalClient, exec
 }
 
 // Records a hedged execution in Redis.
-func RecordHedgedExecution(ctx context.Context, rdb redis.UniversalClient, adResource *digest.ResourceName) error {
+func RecordHedgedExecution(ctx context.Context, rdb redis.UniversalClient, adResource *digest.ResourceName, groupIdForMetrics string) error {
 	key, err := redisKeyForPendingExecutionID(ctx, adResource)
 	if err != nil {
 		return err
 	}
+
 	return rdb.HIncrBy(ctx, key, executionCountKey, 1).Err()
 }
 
+// Records a merged execution.
+func RecordMergedExecution(ctx context.Context, rdb redis.UniversalClient, adResource *digest.ResourceName, groupIdForMetrics string) error {
+	key, err := redisKeyForPendingExecutionID(ctx, adResource)
+	if err != nil {
+		return err
+	}
+
+	if err := rdb.HIncrBy(ctx, key, actionCountKey, 1).Err(); err != nil {
+		return err
+	}
+
+	hash := rdb.HGetAll(ctx, key)
+	if err := hash.Err(); err != nil {
+		log.Debugf("Error reading action-merging state from Redis: %s", err)
+	}
+	recordCountMetric(ctx, hash, groupIdForMetrics)
+	recordSubmitTimeOffsetMetric(ctx, hash, groupIdForMetrics)
+	return nil
+}
+
+func recordCountMetric(ctx context.Context, hash *redis.StringStringMapCmd, groupIdForMetrics string) {
```

Comment:
nit: you could write `hash, err := rdb.HGetAll(ctx, key).Result()` so that `hash` winds up being a plain old `map[string]string` instead of this `*redis.StringStringMapCmd` type

Reply:
done